pkg/apiserver/rest/dualwriter.go

```diff
@@ -80,14 +81,23 @@ const (
 	Mode4
 )
 
-var CurrentMode = Mode2
-
-//TODO: make CurrentMode customisable and specific to each entity
-// change DualWriter signature to get the current mode as an argument
-
 // NewDualWriter returns a new DualWriter.
-func NewDualWriter(legacy LegacyStorage, storage Storage) DualWriter {
-	return selectDualWriter(CurrentMode, legacy, storage)
+func NewDualWriter(kind string, features featuremgmt.FeatureToggles, legacy LegacyStorage, storage Storage) DualWriter {
+	// #TODO replace string with Kind type
+	switch kind {
+	case "playlist":
+		if features.IsEnabledGlobally(featuremgmt.FlagDualWritePlaylistsMode3) {
+			return NewDualWriterMode3(legacy, storage)
+		}
+
```

Comment:
I'm torn on whether to add this yet, since we still need to add the logic to track the current mode and to decide whether it's safe to switch to the requested mode. For mode 1 and mode2 we basically just use the requested mode but for mode3 we will need to verify that all legacy objects have been synced while in mode 2.

If the intent here is just to be able to manually test mode3 behavior I think that's fine, just want to be clear on the intent.

Reply:
I agree that this logic should not live here but rather when we initialise playlist's dualwriter: ideally this fn should be entity agnostic

Reply:
👍  I'll remove the mode 3 part and see how/if to move other things around.